src/Lecture2.hs

```diff
@@ -48,7 +48,9 @@ zero, you can stop calculating product and return 0 immediately.
 84
 -}
 lazyProduct :: [Int] -> Int
-lazyProduct = error "TODO"
+lazyProduct [] = 1
+lazyProduct (0:xs) = 0
```

Comment:
I see a GHC warning that the tail of the list is not used. In that case, you can replace the `xs` variable with `_` 🙂 
```suggestion
lazyProduct (0:_) = 0
```